src/isPrime.js

```diff
@@ -0,0 +1,24 @@
+function isPrime(n) {
+if ( n === 1) {
+  return false;
+}
+if ( n === 2) {
+  return true;
+}
+else
+  {
+    for (let x = 2; x< n/2; x++) {
```

Comment:
The smallest prime factor (if it exists) is _much_ smaller than `n/2`. If `k` is a factor of `n` then `n/k` is also a factor. If `k <= n/k` then we have

```
k * k <= k * (n / k) = n
```

which means that `k * k <= n`, i.e., `k <= sqrt(n)`.

So the smallest factor of `n` can't be larger than `sqrt(n)`.

I wrote this up if you want a more mathy line of reasoning: http://jesse-primes.surge.sh/